lib/galaxy/managers/datatypes.py

```diff
@@ -89,10 +90,43 @@ def view_converters(datatypes_registry: Registry) -> DatatypeConverterList:
     return parse_obj_as(DatatypeConverterList, converters)
 
 
+def _get_edam_details(datatypes_registry: Registry, edam_ids: Dict[str, str]):
+    details_dict = {}
+    for (format, edam_iri) in edam_ids.items():
+        edam_details = datatypes_registry.edam.get(edam_iri, {})
+
+        details_dict[format] = {
+            "prefix_IRI": edam_iri,
+            "label": edam_details.get("label", None),
+            "definition": edam_details.get("definition", None),
+        }
+
+    return details_dict
+
+
+def view_edam_formats(
+    datatypes_registry: Registry, id_only: Optional[bool] = True
+) -> Union[Dict[str, str], DatatypesEDAMDetailsDict]:
+    if id_only:
+        return datatypes_registry.edam_formats
+    else:
+        return _get_edam_details(datatypes_registry, datatypes_registry.edam_formats)
+
+
+def view_edam_data(
```

Comment:
```suggestion
async def view_edam_data(
```

otherwise this will be pushed into a threadpool.

Reply:
Actually all functions should be async here, that slipped through the review process.

Reply:
Also the nested one, or only the top-level functions?

Reply:
Hmm, ideally all the function calls would be async, but that'd break down at parse_obj_as, and in fact at least the pydantic validation is not async ... which, looking back, explains why the FastAPI documentation carefully avoids constructing pydantic models and instead annotates what the response model is. The good news is that I think the use of parse_obj_as is not necessary ... but to keep the PR focused, do you want to make those routes sync for now and we address that in another PR ?

Reply:
Also I'm gonna say this is really difficult to track, maybe FastAPI should have the option to emit a warning when it encounters a sync call in a async route.

Reply:
I'm noticing I didn't use parse_obj_as. Should I keep my routes async and revert the existing ones for now?

Reply:
That sounds good to me!

Reply:
Actually I think I'm totally wrong about this, regular def is fine if you're not doing blocking IO, which you're not.